lib/anyway/config.rb

```diff
@@ -412,13 +445,41 @@ def pretty_print(q)
 
     def validate_required_attributes!
       self.class.required_attributes.select do |name|
-        values[name].nil? || (values[name].is_a?(String) && values[name].empty?)
+        required_name_missing?(name)
       end.then do |missing|
         next if missing.empty?
         raise_validation_error "The following config parameters for `#{self.class.name}(config_name: #{self.class.config_name})` are missing or empty: #{missing.join(", ")}"
       end
     end
 
+    def required_name_missing?(name)
+      return missing_in_environment?(name) if attributes_with_environments.has_key?(name)
+
+      value_missed?(name)
+    end
+
+    def missing_in_environment?(attr)
+      envs = attributes_with_environments[attr]
+      case envs
+      when Array
+        value_missed?(attr) if envs.include?(Anyway::Settings.current_environment)
+      when Hash
+        exclude_key = self.class::ENV_OPTION_EXCLUDE
+        except_envs = envs[exclude_key].flatten
+        value_missed?(attr) unless except_envs.include?(Anyway::Settings.current_environment)
+      else
+        false
+      end
+    end
+
+    def value_missed?(name)
+      values[name].nil? || (values[name].is_a?(String) && values[name].empty?)
+    end
+
+    def attributes_with_environments
+      self.class.send(__method__)
```

Comment:
Why do we use `__method__` here? Do we need this delegation at all (that makes specs fail due to the addition of yet another reserved name, see CI)?